prometheus_client/metrics.py

```diff
@@ -58,6 +58,16 @@ def _is_observable(self):
         # * the child of a labelled metric.
         return not self._labelnames or (self._labelnames and self._labelvalues)
 
+    def _raise_if_not_observable(self):
+        # Functions that mutate the state of the metric, for example incrementing
+        # a counter, will fail if the metric is not observable, because only if a
+        # metric is observable will the value be initialized.
+        if not self._is_observable():
+            raise ValueError(
+                '%s metric is not observable. This is likely because the metric requires '
```

Comment:
Thinking if we want to make this more detailed, for example we could use inspect and get the method name that is being invoked. Also, we may want to use this in other metrics as well. I added the check to counters, but the same issue exists in gauges as well for example

Reply:
I'm not sure "observable" is the right word to use here, it'll confuse users. Missing label values would be more accurate.